.github/workflows/cancel.yml

```diff
@@ -16,7 +16,7 @@
 # under the License.
 
 name: Cancel
-on: [push]
+on: [push, pull_request]
```

Comment:
When does a 'push' event and a 'pull_request' event happen?

AFAIK, no tests are running before a PR is created.

Reply:
In relation to : https://github.com/apache/beam/blob/master/.github/workflows/build_wheels.yml#L21

Push: when new commits are pushed to the specified branches. e.g. new merge commit on master appear. So it is possible to run rests without PR.
Pull request: when PR to specified branch is created and then again when new commits appear in the PR.

It is not critical since PR builds do not delete/upload anything to the GCS (push builds do), but I think it would be convenient to decrease queue by introducing such a small change.

Reply:
> e.g. new merge commit on master appear. So it is possible to run rests without PR.
I do not understand how/when this would happen. Because we do not allow direct pushes to master. And if it is already in master a PR would not be helpful. Maybe this example applies to release branches?

Overall, I am fine with the change. It is ok to cancel tests when a new PR is created.

Reply:
I wasn't aware that direct push to master is not allowed however merge commits which appear on master also trigger _push_ event:
Here you can check filtered wheels builds for _push_ event on _master_:
https://github.com/apache/beam/actions?query=event%3Apush+branch%3Amaster+workflow%3A%22Build+python+wheels%22

and example of _push_ events similar on the release branch `release-2.23.0`:
https://github.com/apache/beam/actions?query=event%3Apush+branch%3Arelease-2.23.0

In comparison here are "_pull request_" events triggered on the various branches:
https://github.com/apache/beam/actions?query=event%3Apull_request